src/Integration/NewConnectedMode/ConfigFileUtilities.cs

```diff
@@ -0,0 +1,98 @@
+/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Alm.Authentication;
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Integration.Helpers;
+using SonarLint.VisualStudio.Integration.Persistence;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarQube.Client.Helpers;
+
+namespace SonarLint.VisualStudio.Integration.NewConnectedMode
+{
+    internal static class ConfigFileUtilities
+    {
+        public static BoundSonarQubeProject ReadBindingFile(string fullConfigFilePath, ICredentialStore credentialStore, ILogger logger, IFile fileWrapper)
```

Comment:
Code extracted from _SolutionBindingSerializer_. Minor changes to make it use the _IFile_ abstraction.

Reply:
[cosmetic] I would split this long line because it is hard to read/review.